packages/ember-routing/lib/system/router.js

```diff
@@ -669,7 +669,7 @@ const EmberRouter = EmberObject.extend(Evented, {
       if (qp) {
         delete queryParams[key];
         queryParams[qp.urlKey] = qp.route.serializeQueryParam(value, qp.urlKey, qp.type);
-      } else {
+      } else if (value !== undefined) {
```

Comment:
Super not-picky but I'd rather leave this as an unconditional else clause. Could you swap this to check if it's undefined and just noop (as an else if before the final else)?